src/combined.cc

```diff
@@ -0,0 +1,253 @@
+
+    // Receiver address
+    Ipv4Address receiver_addr = interfaces_3_5.GetAddress(1);
+
+    // UDP Congestion traffic setup
+    uint16_t udp_port = 5001;
+    OnOffHelper udp_source("ns3::UdpSocketFactory",
+                           InetSocketAddress(receiver_addr, udp_port));
+    udp_source.SetAttribute(
+        "OnTime", StringValue("ns3::ConstantRandomVariable[Constant=1]"));
+    udp_source.SetAttribute(
+        "OffTime", StringValue("ns3::ConstantRandomVariable[Constant=0]"));
+    udp_source.SetAttribute("DataRate", DataRateValue(DataRate("1Mbps")));
+    udp_source.SetAttribute("PacketSize", UintegerValue(1024));
+
+    ApplicationContainer udp_app = udp_source.Install(nodes.Get(0));
+    udp_app.Start(Seconds(3.0));
+    udp_app.Stop(Seconds(6.0));
+
+    // TCP Setup
+    SetupTCPConfig();
+    uint16_t tcp_port = 5002;
+    BulkSendHelper tcp_source("ns3::TcpSocketFactory",
+                              InetSocketAddress(receiver_addr, tcp_port));
+    tcp_source.SetAttribute("MaxBytes", UintegerValue(10000)); // Tweak this
+    ApplicationContainer tcp_app = tcp_source.Install(nodes.Get(1));
+    tcp_app.Start(Seconds(0.0));
+    tcp_app.Stop(Seconds(10.0));
+
+    // Packet sink setup (Receiver node)
+    PacketSinkHelper sink("ns3::TcpSocketFactory",
+                          InetSocketAddress(Ipv4Address::GetAny(), tcp_port));
+    ApplicationContainer sink_app = sink.Install(nodes.Get(5));
+    tcp_app.Start(Seconds(0.0));
+    tcp_app.Start(Seconds(10.0));
+
+    // SimulationQueue::sinkAddress =
+    //     Mac48Address::ConvertFrom(getDevice<1>(devices_3_5)->GetAddress());
+    // NOTE: Is TrafficControlHelper needed here?
+
+    // LFA Alternate Path setup
+    // Set up an alternate forwarding target, assuming you have an alternate
+    // path configured
+
+    // TODO: Need some help with setting alternate target
+    setAlternateTarget<0>(devices_2_3, getDevice<0>(devices_2_4));
+    setAlternateTarget<1>(devices_2_3, getDevice<1>(devices_4_3));
+    // setAlternateTarget<0>(devices01, getDevice<0>(devices02));
+    // setAlternateTarget<0>(devices02, getDevice<0>(devices01));
+
+    // setAlternateTarget<0>(devices12, getDevice<1>(devices01));
+    // setAlternateTarget<1>(devices01, getDevice<0>(devices12));
+
+    // setAlternateTarget<1>(devices02, getDevice<1>(devices12));
+    // setAlternateTarget<1>(devices12, getDevice<1>(devices02));
+
+    // enableRerouting(getQueue<0>(devices_2_3));
+    toggleCongestion(getQueue<0>(devices_0_2));
+    toggleCongestion(getQueue<1>(devices_0_2));
+    toggleCongestion(getQueue<0>(devices_2_4));
+    toggleCongestion(getQueue<1>(devices_2_4));
+    toggleCongestion(getQueue<0>(devices_4_3));
+    toggleCongestion(getQueue<1>(devices_4_3));
+    toggleCongestion(getQueue<0>(devices_3_5));
+    toggleCongestion(getQueue<1>(devices_3_5));
+    toggleCongestion(getQueue<0>(devices_1_2));
+    toggleCongestion(getQueue<1>(devices_1_2));
```

Comment:
some of these devices are not using a simulation queue, and could trigger a segfault.

e.g. `devices_0_2`

Reply:
only `devices_2_3` is has a simulation queue